docs/wiki/TestingUsage.md

```diff
@@ -36,29 +39,88 @@ Invoke-Pester -Configuration @{
 }
 ```
 
-### Handling Resource IDs or Parameters that require or contain Subscription IDs
+## Tool: Use The Test-ModuleLocally Script To Perform Pester Testing, Token Replacement and Deployment of the Module.
 
-- Scenarios where resources have dependencies on other resources, which may require to be linked using `resourceId` references. [Example](../../arm/Microsoft.Network/virtualNetworksResources/virtualNetworkPeerings/.parameters/parameters.json)
+The below example can be used on your local environment to load the 'Test-ModuleLocally.ps1' script and modify the required parameters as below to enable you to perform the tests locally before pushing to source control.
+
+```powershell
+#########[ Function Test-ModulesLocally.ps1 ]#############
+
+. 'C:\Users\user\github\user\ResourceModules\utilities\tools\Test-ModuleLocally.ps1'
+
+# REQUIRED INPUT FOR TESTING
+$TestModuleLocallyInput = @{
+    ModuleName                    = 'Microsoft.Network\applicationSecurityGroups'
+    PesterTest                    = $true
+    DeploymentTest                = $true
+    ValidationTest                = $false
+    ValidateOrDeployParameters    = @{
+        Location          = 'australiaeast'
+        ResourceGroupName = 'validation-rg'
+        SubscriptionId    = '12345678-1234-1234-1234-123456789123'
+        ManagementGroupId = 'mg-contoso'
+        RemoveDeployment  = $false
+    }
+    DeployAllModuleParameterFiles = $false
+    GetParameterFileTokens        = $true
+    #TokenKeyVaultName             = 'contoso-platform-kv'
+    OtherCustomParameterFileTokens      = @(
+        @{ Name = 'deploymentSpId'; Value = '12345678-1234-1234-1234-123456789123' }
+        @{ Name = 'tenantId'; Value = '12345678-1234-1234-1234-123456789123' }
+    )
+}
+
+```
+
+### Handling Parameters that require or contain a value that should be tokenized
+
+The following scenarios are common to when to use a token value in the parameter file. Refer to [Pipeline Design](./PipelinesDesign) for more details.
+
+- Scenarios where resources have dependencies on other resources, which may require to be linked using `resourceId` references.
 
     ```json
     // Example
     "remoteVirtualNetworkId": {
-        "value": "/subscriptions/12345678-abcd-abcd-abcd-12345678abcd/resourceGroups/validation-rg/providers/Microsoft.Network/virtualNetworks/adp-sxx-az-vnet-x-peer01"
+        "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/virtualNetworks/adp-sxx-az-vnet-x-peer01"
```

Comment:
what about the resource group name? should it also be tokenized?

Reply:
Should is a hard word,, but we've tokenized them in our pilot resources (NSG Parameter File). But I applied here as well